x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.test.ts

```diff
@@ -170,67 +214,195 @@ describe('EngineLogic', () => {
         expect(EngineLogic.actions.setEngineData).toHaveBeenCalledWith(mockEngineData);
       });
 
-      it('handles errors', async () => {
+      it('handles 4xx errors', async () => {
         mount();
         jest.spyOn(EngineLogic.actions, 'setEngineNotFound');
-        http.get.mockReturnValue(Promise.reject('An error occured'));
+        http.get.mockReturnValue(Promise.reject({ response: { status: 404 } }));
 
         EngineLogic.actions.initializeEngine();
         await nextTick();
 
         expect(EngineLogic.actions.setEngineNotFound).toHaveBeenCalledWith(true);
       });
+
+      it('handles 5xx errors', async () => {
+        mount();
+        jest.spyOn(EngineLogic.actions, 'setEngineNotFound');
```

Comment:
```suggestion
```

I don't think you need to spy on this action here, it's not being checked below

Reply:
Awesome catch, thank you!! 🦅

Reply:
https://github.com/elastic/kibana/pull/103041/commits/308a8804329fd8beb1a8d63c7570d1d42110a685